### PR TITLE
MCR-3094 add xpath-based classification mapping

### PR DESCRIPTION
**Pull request by erodde**
[Link to jira](https://mycore.atlassian.net/browse/MCR-3094).

### CODE REVIEW

**Comment by erodde** [comment]
I only tested the feature through unit tests so far, as our applications are not yet compatible to 2023.06.x. I refactored org.mycore.oai.classmapping.MCRClassificationMappingEventHandler quite a bit, maybe you could look at it as the original author @rsteph-de ?

**Comment by erodde** [comment]
Test with local MIR, feature branch and PR https://github.com/MyCoRe-Org/mycore-classifications/pull/17 was successfull, both configured XPath-mappings could be generated (from two different publications matching the respective XPath):

`<mods:classification generator="xpathmapping2orcidWorkType-mycore" authorityURI="http://localhost:8291/mir/classifications/orcidWorkType" valueURI="http://localhost:8291/mir/classifications/orcidWorkType#journal-article"/>`

`<mods:classification generator="xpathmapping2orcidWorkType-mycore" authorityURI="http://localhost:8291/mir/classifications/orcidWorkType" valueURI="http://localhost:8291/mir/classifications/orcidWorkType#online-resource"/>`

Adding an XPath-mapping in an exisiting publication through a repair statement works. Subsequent executions of the EventHandler after making changes in a publication work, mappings are not duplicated. This test only considers the `org.mycore.mods.classification.MCRClassificationMappingEventHandler`.

File: mycore-mods/src/test/resources/MCRClassificationMappingEventHandlerTest/orcidWorkType.xml [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+<mycoreclass ID="orcidWorkType" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+             xsi:noNamespaceSchemaLocation="MCRClassification.xsd">
+    <label xml:lang="de" text="ORCID Werktyp"/>
+    <label xml:lang="en" text="ORCID work type"/>
+    <categories>
+        <category ID="journal-article">
+            <label xml:lang="de" text="Zeitschriftenartikel"/>
+            <label xml:lang="en" text="Journal article"
+                   description="Articles in peer-reviewed publications that disseminate the results of original research and scholarship."/>
+            <label xml:lang="x-mapping-xpath" text="mods:genre='article' and mods:relatedItem[@type='host']/mods:genre='journal'" />
```
Comment (kkrebs) [inline mycore-mods/src/test/resources/MCRClassificationMappingEventHandlerTest/orcidWorkType.xml]: I think we should use the typical MIR MODS xpath here

File: mycore-mods/src/test/resources/MCRClassificationMappingEventHandlerTest/testMods2.xml [hunk 2]
```diff
@@ -0,0 +1,15 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<mods:mods xmlns:mods="http://www.loc.gov/mods/v3" xmlns:xlink="http://www.w3.org/1999/xlink">
+    <mods:typeOfResource>text</mods:typeOfResource>
+    <mods:genre authority="test" displayLabel="test">article</mods:genre>
```
Comment (kkrebs) [inline mycore-mods/src/test/resources/MCRClassificationMappingEventHandlerTest/testMods2.xml]: I think we should use the typical MIR MODS xpath here

File: mycore-mods/src/test/resources/MCRClassificationMappingEventHandlerTest/testMods2.xml [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+    <mods:titleInfo xml:lang="de" xlink:type="simple">
+        <mods:title>Test-Titel</mods:title>
+    </mods:titleInfo>
+    <mods:language>
+        <mods:languageTerm authority="rfc5646" type="code">de</mods:languageTerm>
+    </mods:language>
+    <mods:accessCondition type="use and reproduction" xlink:type="simple">cc_by-nc</mods:accessCondition>
+    <mods:relatedItem type="host" xlink:href="host">
+        <mods:genre authority="test" displayLabel="test">journal</mods:genre>
```
Comment (kkrebs) [inline mycore-mods/src/test/resources/MCRClassificationMappingEventHandlerTest/testMods2.xml]: I think we should use the typical MIR MODS xpath here

**Review by kkrebs — CHANGES_REQUESTED** [review]
Works with MIR and looks good for me. I have just one tiny suggestion for improvement, see my comments.

**Review by kkrebs — APPROVED** [review]

**Review by fluetze — APPROVED** [review]

**Pull request merged**